### DISCUSSION TITLE
Что-то я потерял хватку...

### DISCUSSION THREAD

**Discussion by aw-c** [nillerusr/source-engine#332] [Q&A]
Типа я не нашёл никаких компоновщиков и спрашиваю у более прошаренных, что мне дальше делать с получившимися pdb и o файлами, как их перевести в полноценные экзекутбл и длл

**Comment by GreMirid** [comment]
Тебе не нужны исходники Source. Тебе нужна книга по компилируемым языкам и их компиляторам (в частности языка C) и усидчивость в прочтении документаций и факов.
И лишь затем понять, почему твой проект не собрался до конца на твоей конфигурации. (Кстати, какой?)

**Reply by aw-c** [reply]
Windows 10 22H2, компилятор msvc

**Reply by GreMirid** [reply]
Допустим. Тогда я просто хочу в кое что убедиться, у тебя есть директория source-engine\build\out ?

**Reply by aw-c** [reply]
Да просто, её нет

**Reply by GreMirid** [reply]
Лог компиляции, а так же какие параметры ты применял при конфигурации проекта waf можешь скинуть?

**Reply by aw-c** [reply]
https://pastebin.com/Xx96Gz6P

делал сначала waf -T release потом просто waf и он скомпилил o и pdb файлы, линковщик не запустился и я не нашёл где его запускать

**Reply by GreMirid** [reply]
Попробуй вот такую команду:
`waf.bat --64bits configure -T release --prefix=build/out/ --enable-opus --build-games=hl2`

**Reply by GreMirid** [reply]
Вообще тут целый фак по сборке:
https://github.com/nillerusr/source-engine/wiki/Source-Engine-(RU)

**Reply by aw-c** [reply]
Да я по этому факю и следовал, вообще какая-то ебатория. Кстати с твоей командой тоже ничего нет, хотя в логе ошибок нет.

**Reply by GreMirid** [reply]
В том числе и запускал из под Powershell?

**Reply by GreMirid** [reply]
Более того, а саму компиляцию то запускал в Visual Code?

**Reply by aw-c** [reply]
Капец я лалка, да, всё через пауэр шелл

**Reply by GreMirid** [reply]
![image](https://github.com/nillerusr/source-engine/assets/56092546/be9e77a5-c4a4-42d6-b04c-befafaf4462d)
тут есть tasks.json

**Reply by aw-c** [reply]
А вот с этого и стоило начинать. По таскам из VSC запустилось, спс

**Reply by GreMirid** [reply]
Ну стоило Начать с того, что проект ты только Сконфигурировал, но собрать - не собирал.

**Reply by aw-c** [reply]
Компил произошёл, линковка не произошла

**Comment by aw-c** [comment]
Тогда у меня есть следующий вопрос.
Игра скомпилирована разработчиками с использованием следующих флагов:
`Counts: Pre-VC++ 11.00=53, C/C++=713, /GS=326, /sdl=0, guardN=278`
А эта при дефолтных настройках компилирует со следующими:
`Counts: Pre-VC++ 11.00=0, C/C++=686, /GS=686, /sdl=0, guardN=272`

Как я могу добиться тех же настроек компиляции, если мне нужно учитывать, что на выходе я хочу получить те же сигнатуры функций, что уже имеются в их DLL?

**Reply by nsherbina1999** [reply]
You need to find these flags and manually add them to wscript file for building. I guess it should work then.
